### ISSUE TITLE
HTTPS (secure) redirects to HTTP (insecure)

### ISSUE THREAD

**Issue by edwardcasbon** [syntaxhighlighter/syntaxhighlighter#442]
When requesting https://alexgorbatchev.com/pub/sh/current/scripts/shCore.js over a secure connection (HTTPS) it redirects to http://agorbatchev.typepad.com/pub/sh/3_0_83/scripts/shCore.js, which is insecure, and fails to load on my website which is served over HTTPS.

Please could you update the redirect so that it redirects using the protocol it's requested in?

Many thanks.

**Comment by jmaguilar** [comment]
Same here, it would be great to update these redirections. Thanks!

Anyway, as a workaround, could we link directly to the files at http://agorbatchev.typepad.com? Or there exist another HTTPS link to the current version?

**Comment by edwardcasbon** [comment]
So, to get around the problem I updated my project to get the assets from `https://cdnjs.cloudflare.com/ajax/libs/SyntaxHighlighter/3.0.83/`.

**Comment by jmaguilar** [comment]
I did the same and it works fine, but the problem with this approach is that we are linking to a specific version, not the latest/current version.

**Comment by edwardcasbon** [comment]
Yep, that's the sacrifice I had to make to ensure it worked on our project.

**Comment by AllenConway** [comment]
So I found this SO post where Alex explains the redirect. Even though you link to 'https,' it's actually doing a 302 and directing back to the 'http' version. https://stackoverflow.com/questions/44866175/why-is-syntaxhighlighter-being-fetched-via-http-from-blogger-com 

The problem I had in the past using that cdnjs.cloudflare.com was I pulled up my blog that uses SyntaxHighlighter at a client, and that domain was flagged as insecure and a security risk. I had IT bearing down on me quickly as the traffic sniffed out was from that domain. Might need to look into uploading to a secure location and using my own links if possible.
